bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -82,16 +89,115 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 if (EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL.equals(channelId)
                         && command instanceof DecimalType) {
                     double newTemp = ((DecimalType) command).doubleValue();
-                    if (newTemp == CANCEL_SET_POINT_OVERRIDE) {
+                    // Get a local copy of schedule when needed, saves on api calls
+                    DailySchedules schedule = getDailySchedules();
+                    double expectedTemp = getCurrentSetPoint(schedule);
+                    if (newTemp == CANCEL_SET_POINT_OVERRIDE || newTemp == expectedTemp) {
```

Comment:
No, this checks against the programmed schedule. So expected temp is the expected temp according to the schedule (`getCurrentSetPoint(schedule);` note the reference to the schedule not the zone)

What this means is if you are in a schedule of 2 hours at 15oC, but overriden to 20oC, when you then send 15oC it will process as a cancellation of the override, not an override to 15oC. Results in a much clearer interface to the user rather than "why do i have an override to the same value as the schedule" 

Sending 20oC twice will just end up with the override to 20oC being set twice